Split CalculationValuesMap type into its own files

weinig opened the pull request:
#### b70a65cd663260deccdc9f2f1ecd53b9ad730de1
<pre>
Split CalculationValuesMap type into its own files
<a href="https://bugs.webkit.org/show_bug.cgi?id=295536">https://bugs.webkit.org/show_bug.cgi?id=295536</a>

Reviewed by Darin Adler.

This will let it be used from multiple places.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/Length.cpp:
* Source/WebCore/platform/calc/CalculationValueMap.cpp: Added.
* Source/WebCore/platform/calc/CalculationValueMap.h: Added.

Canonical link: <a href="https://commits.webkit.org/297111@main">https://commits.webkit.org/297111@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/64fd9edffab6d975a55b16ee630baba2e1bbe521

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110539 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30198 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20631 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116565 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60806 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112502 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30877 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38786 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84059 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113487 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24649 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99535 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64500 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24010 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17675 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60360 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94031 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17734 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119355 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37579 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27896 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93024 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37953 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95806 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92847 "Found 2 new API test failures: /WebKitGTK/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/new-window-policy, /WebKitGTK/TestResources:/webkit/WebKitWebResource/get-data-error (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23667 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37853 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15595 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33552 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37474 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42945 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37137 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40476 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38845 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4b9793adda025831d6d5e77379b8f6d17a759473. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47675

weinig commented:
@rniwa neither of the two files noted by the checker, platform/graphics/DisplayRefreshMonitorManager.cpp, platform/mediastream/Mac/CoreAudioCaptureSource.h, are touched by this change.

Are they the fault of this change and I just don't understand? If its just unified build changing what files get compiled together, how can we resolve this problem?

darinadler commented on Source/WebCore/platform/calc/CalculationValueMap.h:
```diff
@@ -0,0 +1,109 @@
+        Entry(Ref<CalculationValue>&&);
+    };
+
+    unsigned m_nextAvailableHandle;
```
```suggestion
    unsigned m_nextAvailableHandle { 1 };
```

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/087aaa847894febc416beb283bbfe93ce39a27b2. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47675

darinadler commented on Source/WebCore/platform/calc/CalculationValueMap.h:
```diff
@@ -0,0 +1,108 @@
+public:
+    static CalculationValueMap& calculationValues();
+
+    CalculationValueMap();
```
I guess if you take my suggestion from earlier, then the default constructor will suffice. Maybe we want the constructor and destructor to both be `private`, though?
```suggestion
    CalculationValueMap() = default;
```

darinadler commented on Source/WebCore/platform/calc/CalculationValueMap.h:
```diff
@@ -0,0 +1,108 @@
+{
+}
+
+inline CalculationValueMap::CalculationValueMap()
+    : m_nextAvailableHandle(1)
+{
+}
+
```
```suggestion
```

weinig replied:
I can also move it to cpp file, since the only caller is also in that cpp file.

weinig committed:
Split CalculationValuesMap type into its own files
https://bugs.webkit.org/show_bug.cgi?id=295536

Reviewed by Darin Adler.

This will let it be used from multiple places.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/Length.cpp:
* Source/WebCore/platform/calc/CalculationValueMap.cpp: Added.
* Source/WebCore/platform/calc/CalculationValueMap.h: Added.

Canonical link: https://commits.webkit.org/297111@main
